detox/src/android/webExpect.js

```diff
@@ -0,0 +1,309 @@
+  }
+}
+
+class WebClearTextAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.clearText(element._call);
+  }
+}
+
+class WebScrollToViewAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.scrollToView(element._call);
+  }
+}
+
+class WebGetTextAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.getText(element._call);
+  }
+}
+
+class WebRunScriptAction extends WebAction {
+  constructor(element, script) {
+    super();
+    this._call = WebElementApi.runScript(element._call, script);
+  }
+}
+
+class WebRunScriptWithArgsAction extends WebAction {
+  constructor(element, script, args) {
+    super();
+    this._call = WebElementApi.runScriptWithArgs(element._call, script, args);
+  }
+}
+
+class WebGetCurrentUrlAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.getCurrentUrl(element._call);
+  }
+}
+
+class WebGetTitleAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.getTitle(element._call);
+  }
+}
+
+class WebFocusAction extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.runScript(element._call, focus);
+  }
+}
+
+class WebSelectAllText extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.runScript(element._call, selectElementContents);
+  }
+}
+
+class WebMoveCursorEnd extends WebAction {
+  constructor(element) {
+    super();
+    this._call = WebElementApi.runScript(element._call, moveCursorToEnd);
+  }
+}
+class WebViewElement {
+  constructor(invocationManager, deviceDriver, emitter, matcher) {
+    this._invocationManager = invocationManager;
+    this._deviceDriver = deviceDriver;
+    this._emitter = emitter;
+    this._matcher = matcher;
+
+    if (matcher !== undefined) {
+      this._call = invoke.callDirectly(EspressoWebDetoxApi.getWebView(matcher._call.value));
+    } else {
+      this._call = invoke.callDirectly(EspressoWebDetoxApi.getWebView());
+    }
+
+    this.element = this.element.bind(this);
+  }
+
+  element(webMatcher, index = 0) {
+    return new WebElement(this._invocationManager, this._deviceDriver, this, webMatcher, index);
+  }
+}
+
+class WebElement {
+  constructor(invocationManager, deviceDriver, webViewElement, matcher, index) {
+    this._invocationManager = invocationManager;
+    this._deviceDriver = deviceDriver;
+    this._call = invoke.callDirectly(WebViewElementApi.element(webViewElement._call, matcher._call.value, index));
+  }
+
+  // At the moment not working on content-editable
+  async tap() {
+    return await new ActionInteraction(this._invocationManager, new WebTapAction(this)).execute();
+  }
+
+  async typeText(text, isContentEditable = false) {
+    if (isContentEditable) {
+      return await this._deviceDriver.typeText(text);
```

Comment:
This won't work without `deviceId` ☢️ !! @OrenZak , are you sure it worked before?

Reply:
Fixed, but I've got another question – whether it is justified to hide the workaround that way?
The non-standard second parameter makes a bit concerned, although it is a minor problem still.

Reply:
@noomorph this appears to be my fault. This used to use the `device`, but I didn't like having a top-level API in the form of `device.typeText()`. Looks like we'll have to revisit.

Reply:
Well I can't say I am totally against that API. There can be scenarios with hardware keyboard, where you are sending keystorkes, why not?

Reply:
It's more about the inconsistency of the API. We currently allow for text-typing strictly in the context of an element (i.e. it is an element interaction function). I'm not sure what the scope is - of just typing things generically, regardless of the keyboard type. Also, I'm not sure there is an iOS equivalent.